pkg/agent/externalnode/external_node_controller_linux_test.go

```diff
@@ -0,0 +1,1155 @@
+		IPs:           []net.IP{net.ParseIP("10.5.6.9")},
+		OVSPortConfig: &interfacestore.OVSPortConfig{OFPort: 2},
+		EntityInterfaceConfig: &interfacestore.EntityInterfaceConfig{
+			EntityName:      "externalEntity",
+			EntityNamespace: "externalEntityNamespace",
+			UplinkPort: &interfacestore.OVSPortConfig{
+				PortUUID: ifaceUUID2,
+				OFPort:   6,
+			},
+		},
+	}
+	if3 = interfacestore.InterfaceConfig{
+		InterfaceName: ifaceName1,
+		IPs:           []net.IP{net.ParseIP("10.5.6.9")},
+		OVSPortConfig: &interfacestore.OVSPortConfig{OFPort: 1, PortUUID: portUUID1},
+		EntityInterfaceConfig: &interfacestore.EntityInterfaceConfig{
+			EntityName:      "externalEntity",
+			EntityNamespace: "externalEntityNamespace",
+			UplinkPort: &interfacestore.OVSPortConfig{
+				PortUUID: ifaceUUID1,
+				OFPort:   6,
+			},
+		},
+	}
+	portData1 = &ovsconfig.OVSPortData{
+		Name: "portName",
+		ExternalIDs: map[string]string{
+			ovsExternalIDUplinkName:      "uplinkName",
+			ovsExternalIDUplinkPort:      portUUID2,
+			ovsExternalIDEntityName:      "externalEntity",
+			ovsExternalIDEntityNamespace: "externalEntityNamespace",
+			ovsExternalIDIPs:             "10.22.33.44,aabb:1122::101:102",
+		},
+	}
+)
+
+func TestUpdateExternalNode(t *testing.T) {
+	defer mockRemoveLinkRoutes(nil)()
+	defer mockRemoveLinkIPs(nil)()
+	defer mockLinkSetUp(nil)()
+	defer mockLinkSetMTU(nil)()
+
+	defer mockRenameInterface(nil)()
+	defer mockHostInterfaceExists(true)()
+	defer mockConfigureLinkAddresses(nil)()
+	iface := &net.Interface{
+		Index:        1,
+		HardwareAddr: net.HardwareAddr{0x22, 0x33, 0x44, 0x55, 0x66, 0x77, 0x88},
+		MTU:          1500,
+	}
+	_, addr1, _ := net.ParseCIDR("10.20.30.40")
+	defer mockGetInterfaceConfig(
+		[]mockGetInterfaceConfigParam{
+			{iface, []*net.IPNet{addr1}, nil},
+		},
+	)()
+	defer mockConfigureLinkRoutes(nil)()
+	externalNode1 := v1alpha1.ExternalNode{
+		ObjectMeta: metav1.ObjectMeta{Name: "if1", Namespace: "ns1", Labels: map[string]string{"en": "if1"}},
+		Spec: v1alpha1.ExternalNodeSpec{
+			Interfaces: []v1alpha1.NetworkInterface{{IPs: []string{"10.5.6.9"}}},
+		},
+	}
+	externalNode2 := v1alpha1.ExternalNode{
+		ObjectMeta: metav1.ObjectMeta{Name: "if2", Namespace: "ns1", Labels: map[string]string{"en": "if1"}},
+		Spec: v1alpha1.ExternalNodeSpec{
+			Interfaces: []v1alpha1.NetworkInterface{{IPs: []string{"10.5.6.8"}}},
+		},
+	}
+	externalNode3 := v1alpha1.ExternalNode{
+		ObjectMeta: metav1.ObjectMeta{Name: "if1", Namespace: "ns1", Labels: map[string]string{"en": "if1"}},
+		Spec: v1alpha1.ExternalNodeSpec{
+			Interfaces: []v1alpha1.NetworkInterface{{IPs: []string{"10.5.6.9"}}},
+		},
+	}
+	for _, tt := range []struct {
+		name               string
+		preEN              *v1alpha1.ExternalNode
+		curEN              *v1alpha1.ExternalNode
+		preIf              *interfacestore.InterfaceConfig
+		curIf              *interfacestore.InterfaceConfig
+		interfaceAction    string
+		syncedExternalNode *v1alpha1.ExternalNode
```

Comment:
Please keep the name format consistent. better to `preEN` and `curEN` to `preExternalNode` and `curExternalNode` for readability.

Reply:
updated